rotkehlchen/db/drivers/server.py

```diff
@@ -0,0 +1,246 @@
+import os
+import signal
+import sqlite3
+import sys
+from collections import defaultdict
+from collections.abc import Sequence
+from dataclasses import dataclass
+from enum import Enum, StrEnum, auto
+from types import FrameType
+from typing import Any, Final, TypeAlias, TypedDict
+
+import zmq
+from pysqlcipher3 import dbapi2 as sqlcipher
+
+from rotkehlchen.args import app_args
+from rotkehlchen.utils.mixins.enums import SerializableEnumNameMixin
+
+UnderlyingCursor: TypeAlias = sqlite3.Cursor | sqlcipher.Cursor  # pylint: disable=no-member
+UnderlyingConnection: TypeAlias = sqlite3.Connection | sqlcipher.Connection  # pylint: disable=no-member
+
+
+class DBConnectionType(Enum):
+    USER = auto()
+    TRANSIENT = auto()
+    GLOBAL = auto()
+
+
+DB_CONNECTION_ADDRESS: Final = 'tcp://localhost'
+DEFAULT_DB_WRITER_PORT: Final = 5555
+
+
+class SerializedZMQCallData(TypedDict):
+    """Dictionary type for serialized ZMQ call data"""
+    connection_type: int
+    db_path: str
+    cursor_name: str | None
+    method: str
+    args: Sequence[Any]
+    kwargs: dict[str, Any]
+
+
+class SerializedDBWriteResult(TypedDict):
+    """Dictionary type for serialized DB write result"""
+    rowcount: int
+    lastrowid: int
+
+
+class SerializedZMQReturnData(TypedDict):
+    """Dictionary type for serialized ZMQ return data"""
+    result: SerializedDBWriteResult | None
+    error: Any
+
+
+class DbMethod(StrEnum, SerializableEnumNameMixin):
+    INITIALIZE = auto()
+    OPEN_CURSOR = auto()
+    CLOSE_CURSOR = auto()
+    EXECUTE = auto()
+    EXECUTEMANY = auto()
+    EXECUTESCRIPT = auto()
+    ROLLBACK = auto()
+    COMMIT = auto()
+    CLOSE = auto()
+
+
+@dataclass
+class ZMQCallData:
+    """The data packet sent through ZMQ call"""
+    connection_type: DBConnectionType
+    db_path: str
+    cursor_name: str | None
+    method: DbMethod
+    args: Sequence[Any]
+    kwargs: dict[str, Any]
+
+    def serialize(self) -> SerializedZMQCallData:
+        return {
+            'connection_type': self.connection_type.value,
+            'db_path': self.db_path,
+            'cursor_name': self.cursor_name,
+            'method': self.method,
+            'args': self.args,
+            'kwargs': self.kwargs,
+        }
+
+    @classmethod
+    def deserialize(cls, data: SerializedZMQCallData) -> 'ZMQCallData':
+        return cls(
+            connection_type=DBConnectionType(data['connection_type']),
+            db_path=data['db_path'],
+            cursor_name=data['cursor_name'],
+            method=DbMethod.deserialize(data['method']),
+            args=data['args'],
+            kwargs=data['kwargs'],
+        )
```

Comment:
this method cna raise deserialization error and valueerror afaik. Are those handled? State it in the docstring